optimade/server/entry_collections/entry_collections.py

```diff
@@ -1,34 +1,87 @@
-from abc import abstractmethod
-from typing import Collection, Tuple, List
+from abc import abstractmethod, ABC
+from typing import Tuple, List, Union
+
+from fastapi import HTTPException
+from lark import Transformer
 
-from optimade.server.mappers import BaseResourceMapper
 from optimade.filterparser import LarkParser
 from optimade.models import EntryResource
-from optimade.server.query_params import EntryListingQueryParams
+from optimade.server.config import CONFIG
+from optimade.server.exceptions import BadRequest
+from optimade.server.mappers import BaseResourceMapper
+from optimade.server.query_params import EntryListingQueryParams, SingleEntryQueryParams
+
 
+class EntryCollection(ABC):
+    """ Backend agnostic base class for collections of EntryResources. """
```

Comment:
I don't know if it can find the classes through the `__init__` imports, but it's worth a try to avoid lengthening the doc string.
```suggestion
    """ Backend-agnostic base class for collections of [`EntryResource`][optimade.models.EntryResource]s. """
```

Reply:
As stated below, this kind of link doesn't seem to work; in sphinx I would do :class:`EntryResource`, is there a mkdocs equivalent?

Reply:
There is this one :) But again, #430 will make sure it actually works ... Although it should sort of work already, since I believe `EntryResource` already exists in the API reference?